crates/vm/src/arch/integration_api.rs

```diff
@@ -360,6 +360,18 @@ where
     fn width(&self) -> usize {
         self.adapter.width() + self.core.width()
     }
+
+    fn columns(&self) -> Option<Vec<String>> {
+        let combined_columns = self
+            .adapter
+            .columns()
+            .into_iter()
+            .flatten()
+            .chain(self.core.columns().into_iter().flatten())
+            .collect::<Vec<_>>();
+
+        (!combined_columns.is_empty()).then_some(combined_columns)
```

Comment:
```suggestion
        if let (Some(adapter_columns), Some(core_columns)) = (self.adapter.columns(), self.core.columns()) {
            Some(adapter_columns.into_iter().chain(core_columns).collect())
        } else {
            None
        }
```

I think something like this (haven't tested it) would be more correct: We want to return `Some(_)` iff. both the core and adapter return `Some(_)`.